src/animations.ts

```diff
@@ -0,0 +1,594 @@
+// -- start tsdoc imports --
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { BattleAnim } from "./data/battle-anims";
+// -- end tsdoc imports --
+import { getFrameMs, randGauss, randInt } from "#app/utils";
+import { PokeballType } from "#enums/pokeball";
+import type { Variant } from "./data/variant";
+import type BattleScene from "./battle-scene";
+
+/**
+ * Class for handling general animations such as particle effects.
+ * For battle animations, see {@linkcode BattleAnim}.
```

Comment:
```suggestion
 * For battle animations, see {@linkcode import("./data/battle-anims").BattleAnim}.
```
Maybe use that instead of the 
```ts
// -- start tsdoc imports --
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { BattleAnim } from "./data/battle-anims";
// -- end tsdoc imports --
```
?

Reply:
I've tried this before and it does not work
<img width="665" alt="Screenshot 2025-01-09 at 5 10 29 AM" src="https://github.com/user-attachments/assets/2da5f999-6172-458e-bb04-7e002ac3d25e" />

Reply:
That's dumb

Reply:
I've been wondering about these imports for doc purposes in the way they've been added in a few recent PRs. Could these be impacting execution order and cause more of some of the issues we've had with that? We've been trying to cut circular dependencies too, how do these work in regards to that?
Are they just fully ignored when building because they are not used?

Reply:
@MokaStitcher yes the will indeed influence the build since the file itself will be imported!
That's why I'm so eager to eliminate it. 
BUT if I'm not mistaken using a `type` import `import type { [...]` might eliminate this issue since type imports are for development/compiler only

Reply:
Interesting
Yeah if they can influence the build I'm not sure I'd want them added like that either

Reply:
> BUT if I'm not mistaken using a type import import type { [...] might eliminate this issue since type imports are for development/compiler only

yes type import eliminates the issue (I didnt notice I wasnt using one, nice catch)

Reply:
updated